packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -233,14 +246,21 @@ function findRoutesAndCreateOptimalPath(
                 opts.outputAmountPerEth,
                 opts.inputAmountPerEth,
                 fees,
-            )[0];
-            // NOTE: For Limit/RFQ orders we are done here. No need to scale output
-            adjustedFills.push(nativeFill);
+            )[0] as Fill | undefined;
+            // Note: If the order has an adjusted rate of less than or equal to 0 it will be skipped
+            // and nativeFill will be `undefined`
+            if (nativeFill) {
+                // NOTE: For Limit/RFQ orders we are done here. No need to scale output
+                adjustedFills.push({ ...nativeFill, sourcePathId: sourcePathId ?? hexUtils.random() });
```

Comment:
How can `sourcePathId` be nullish here? I think it's probably fine to just always do `hexUtils.random()` here though.

Reply:
Yeah, it should always be defined here since we are zipping arrays of the same length, but TS doesn't know that. I figured we might as well fall back to generating a random id rather than overriding the types and risk future changes breaking that invariant 😄